caddyprom_test.go

```diff
@@ -0,0 +1,119 @@
+package caddyprom_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	//"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+
+	. "github.com/hairyhenderson/caddyprom"
+)
+
+// This tests the different syntax options of the prometheus directive
+func TestUnmarshalCaddyfileSyntax(t *testing.T) {
+
+	testCaddyfiles := []string{
+		`
+			prometheus
+		`,
+		`
+			prometheus 0.0.0.0:1337
+		`,
+		`
+			prometheus {
+				address 0.0.0.0:1337
+			}
+		`,
+		`
+			prometheus {
+				address 0.0.0.0:1337
+				path /otherpath
+			}
+		`,
+	}
+
+	// metrics struct with default values
+	// TODO replace with actual metrics struct provisioned by caddy
+	metrics := Metrics{
```

Comment:
Shouldn't this be a `*Metrics`?

```suggestion
	metrics := &Metrics{
```